apps/user_ldap/lib/Connection.php

```diff
@@ -611,7 +617,7 @@ private function establishConnection() {
 			//if LDAP server is not reachable, try the Backup (Replica!) Server
 			if ($isBackupHost || $isOverrideMainServer) {
```

Comment:
And make this condition more explicit

```suggestion
			if (($isBackupHost || $isOverrideMainServer) && !shouldUseBackgroundHost) {
```